lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -0,0 +1,336 @@
+            value = str(key_value_pair[1])
+            format_parameters.append({'ParameterName': key, 'ParameterValue': value})
+    else:
+        all_parameters = True
+
+    try:
+        response = conn.reset_cache_parameter_group(CacheParameterGroupName=name, ParameterNameValues=format_parameters, ResetAllParameters=all_parameters)
+    except boto.exception.ClientError as e:
+        module.fail_json(msg=e.message)
+
+    # determine changed
+    new_parameters_dict = make_current_modifiable_param_dict(conn, name)
+    changed = check_changed_parameter_values(values, old_parameters_dict, new_parameters_dict)
+
+    return response, changed
+
+def get_info(conn, name):
+    """ Gets info about the ElastiCache parameter group. Returns false if it doesn't exist or we don't have access. """
+    try:
+        data = conn.describe_cache_parameters(CacheParameterGroupName=name)
+        return data
+    except botocore.exceptions.ClientError as e:
+        return False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        group_family    = dict(required=False, type='str', choices=['memcached1.4', 'redis2.6', 'redis2.8', 'redis3.2']),
+        name            = dict(required=True, type='str'),
+        description     = dict(required=False),
+        state           = dict(required=True),
+        values          = dict(required=False, type='list'),
+    )
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto required for this module')
+
+    parameter_group_family  = module.params.get('group_family')
+    parameter_group_name    = module.params.get('name')
+    group_description       = module.params.get('description')
+    state                   = module.params.get('state')
+    values                  = module.params.get('values')
+
+    # Retrieve any AWS settings from the environment.
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg = str("Either region or AWS_REGION or EC2_REGION environment variable or boto config aws_region or ec2_region must be set."))
+
+    connection = boto3_conn(module, conn_type='client',
+                            resource='elasticache', region=region,
+                            endpoint=ec2_url, **aws_connect_kwargs)
+
+    exists = get_info(connection, parameter_group_name)
+
+    # check that the needed requirements are available
+    if state == 'present' and exists and not values:
+        module.fail_json(msg="Group already exists. Please specify values to modify using the 'values' option or use the state "
+                         "'reset' if you want to reset all or some group parameters.")
+    elif state == 'present' and not exists and not (parameter_group_family or group_description):
+        module.fail_json(msg="Creating a group requires a group name, the group family, and a description.")
```

Comment:
You don't need to check the group name here, since that's a required parameter Ansible will fail before this executes.